src/stepfunctions/steps/utils.py

```diff
@@ -45,3 +46,24 @@ def get_aws_partition():
             return cur_partition
 
     return cur_partition
+
+
+def merge_dicts(first, second, first_name, second_name):
+    """
+    Merges first and second dictionaries into the first one.
+    Values in the first dict are updated with the values of the second one.
+    """
+    if all(isinstance(d, dict) for d in [first, second]):
+        for key, value in second.items():
+            if key in first:
+                if isinstance(first[key], dict) and isinstance(second[key], dict):
+                    merge_dicts(first[key], second[key], first_name, second_name)
+                elif first[key] is value:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The `is` operator performs an identity comparison that uses the built-in `is()` method to test if the compared items are the same object or  not. It should be used with only singletons, such as `None`. We recommend that you use the `==` or `!=` equality operators to determine if two objects are equal.

[Learn more](https://docs.python.org/3/reference/datamodel.html#object.eq)

Reply:
Good point! Will make the change to use `==` in the next commit